lib/restbase.js

```diff
@@ -307,6 +318,17 @@ RESTBase.prototype._request = function (req) {
             self.metrics.endTiming([statName + statusClass, statName + 'ALL'], startTime);
             throw err;
         });
+        if (match.permissions
```

Comment:
I'm wondering if we should propagate permissions down to nested requests, so that we can use that information to a) apply it to nested requests, and b) use it to implicitly forward cookies in outgoing requests (in `defaultWebRequestHandler`).